tests/tar/BUILD

```diff
@@ -60,6 +60,20 @@ genrule(
     cmd = "for i in $(OUTS); do echo 1 >$$i; done",
 )
 
+genrule(
+    name = "generate_dir_file",
+    outs = ["lib"],
+    cmd = "mkdir -p $@; echo 1 >$@/nsswitch.conf",
```

Comment:
This is not a case we should actively support. It doesn't work in remote build situations.

Reply:
Well the behavior is at least currently inconsistent with zip files so not sure what best option here is.

Reply:
Does this [ctx.actions.declare_directory](https://github.com/SeleniumHQ/selenium/pull/11392) based rule look correct?

Reply:
That looks mostly right. It does a declare_directory, then passes the .path of that to some tool to fill in.